rofl-scheduler/src/state.rs

```diff
@@ -0,0 +1,103 @@
+use std::{collections::BTreeMap, time::Instant};
+
+use base64::{prelude::BASE64_STANDARD, Engine};
+use oasis_runtime_sdk::types::address::Address;
+use oasis_runtime_sdk_rofl_market::types::{Deployment, Instance};
+
+use crate::types::{Permissions, METADATA_KEY_PERMISSIONS};
+
+/// Instance state.
+#[derive(Default, Debug, Clone)]
+pub struct InstanceState {
+    /// Address of the instance administrator.
+    admin: Address,
```

Comment:
It is a bit strange to have a default admin address. I prefer having a constructor with admin and permissions as parameters.